torch/csrc/autograd/python_variable_indexing.cpp

```diff
@@ -42,15 +43,15 @@ Py_ssize_t THPVariable_length(PyObject* self) {
 // and tuples of those types. We also handle bools as if they were a
 // Variable[ByteTensor].
 
-static int64_t count_specified_dimensions(PyObject* index) {
+static inline int64_t count_specified_dimensions(PyObject* index) {
   // Count the number of indexed dimensions (everything but ellipsis and None)
   int64_t count = 0;
-  auto size = PyTuple_GET_SIZE(index); // NOLINT(cppcoreguidelines-pro-type-cstyle-cast)
-  for (Py_ssize_t i = 0; i < size; i++) {
+  for (Py_ssize_t i = 0; i < PyTuple_GET_SIZE(index); i++) {
```

Comment:
This looks like a pessimization. If the compiler cannot tell that `PyTuple_GET_SIZE` stays constant across the loop, it cannot hoist it out. Then you are repeatedly doing some complicated query to get size from Python C API.

Reply:
(If it's a macro, all may be well in the end, but it's better if you just make it so the compiler cannot go wrong.)